### ISSUE TITLE
Support Major Metrics

### ISSUE THREAD

**Issue by rejasupotaro** [rejasupotaro/master-thesis#8]
## Overview

According to the list of neural information retrieval (https://github.com/rejasupotaro/master-thesis/issues/5), NDCG and MAP are the common metrics. In this issue, I support these metrics.

### NDCG

- `gain_fn`: A method to calculate gain parameters used in the definitions of the DCG and NDCG metrics, where the input is the relevance label of the item. The gain is often defined to be of the form 2^label-1.
- `rank_discount_fn`: (function) The rank discount function. Note that this implementation of NDCG assumes that this function is *decreasing* as a function of its input.

```python
_DEFAULT_GAIN_FN = lambda label: tf.pow(2.0, label) - 1
_DEFAULT_RANK_DISCOUNT_FN = lambda rank: tf.math.log(2.) / tf.math.log1p(rank)
```

- [model-analysis/ndcg.py at master · tensorflow/model-analysis](https://github.com/tensorflow/model-analysis/blob/master/tensorflow_model_analysis/metrics/ndcg.py)
- [ranking/metrics_impl.py at 6c55d0bc87c761448f1937f46998039c8dbfe280 · tensorflow/ranking](https://github.com/tensorflow/ranking/blob/6c55d0bc87/tensorflow_ranking/python/metrics_impl.py)
- [MatchZoo/normalized_discounted_cumulative_gain.py at 7f6c30d8d80aeab107003ba9357951413d530402 · NTMC-Community/MatchZoo](https://github.com/NTMC-Community/MatchZoo/blob/7f6c30d8d80aeab107003ba9357951413d530402/matchzoo/metrics/normalized_discounted_cumulative_gain.py)

## Links

- [sklearn.metrics.dcg_score — scikit-learn 0.23.1 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.dcg_score.html)
